### DISCUSSION TITLE
Multiple lock files feature doesn't seems to work for VCS dependencies

### DISCUSSION THREAD

**Discussion by liran-cohen-hs** [pdm-project/pdm#1815] [Q&A]
Trying to use this feature for developers to be able to use inner dependencies from local version
didn't try it with regular dependency (from pypi)
Tried 2 different methods:

1 - using PDM add commands, the second one always seems to fail
running:
`pdm add "git+ssh://git@github.com/HiredScore/gimme.git@X.Y.Z#egg=hs_gimme"`
then
`pdm add -e ../gimme --dev`
results in:

> Adding group dev to lockfile
> Adding packages to dev dev-dependencies: hs-gimme @ file:///${PROJECT_ROOT}/../gimme
> 🔒 Lock failed
> Unable to find a resolution for hs-gimme
> because of the following conflicts:
>   hs-gimme @ file:///${PROJECT_ROOT}/../gimme (from project)
>   hs-gimme @ git+ssh://git@github.com/HiredScore/gimme.git@X.Y.Z (from project)

2 - using one PDM add command and the second dependency I've put manually
in this case, the dev dependency is ignored
running: 
`pdm add "git+ssh://git@github.com/HiredScore/gimme.git@X.Y.Z#egg=hs_gimme"`
then, adding this lines manually to pyproject.toml:
```
[tool.pdm.dev-dependencies]
dev = ["hs-gimme @ file:///${PROJECT_ROOT}/../gimme"]
```
and running `pdm lock` is ignoring the dev dependency I added, in addition, any `pdm add` command will now fail with the same resolution error from step 1

referring to this feature:
https://pdm.fming.dev/latest/usage/dependency/#specify-the-lockfile-to-use

I am on the latest PDM version (2.5.0b0)

This is a blocker for us in migrating to PDM, as this feature of developing locally with editable option is widely used across the company

Thanks

**Comment by frostming** [comment]
Because you are adding two different specifications for the same package, you need to make them in separate lockfiles or otherwise one will override the other. Follow this steps:

```bash
pdm add "git+ssh://git@github.com/HiredScore/gimme.git@X.Y.Z#egg=hs_gimme"
# This will create a lockfile named pdm.lock that only contains "default" group
pdm add -e ../gimme --dev -L pdm.dev.lock
# This will create a lockfile named pdm.dev.lock that contains the editalbe local copy

# To install the default version:
pdm install
# Or to install for development:
pdm install -L pdm.dev.lock
```

**Reply by liran-cohen-hs** [reply]
tried that now
same output
```
pdm add -e $HSCODE/gimme --dev -L pdm.dev.lock
Adding packages to dev dev-dependencies: -e file:///Users/liran/hiredscore/git/gimme#egg=hs-gimme
🔒 Lock failed
Unable to find a resolution for hs-gimme
because of the following conflicts:
  -e file:///Users/liran/hiredscore/git/gimme#egg=hs-gimme (from project)
  hs-gimme @ git+ssh://git@github.com/HiredScore/gimme.git@63.4.65 (from project)
```

**Reply by frostming** [reply]
There was a bug to be released in 2.5.0. Please try it again on the next stable release.

**Reply by liran-cohen-hs** [reply]
> There was a bug to be released in 2.5.0. Please try it again on the next stable release.

thanks ! any time estimation on when it will be released ?

**Reply by frostming** [reply]
NOW!

**Reply by liran-cohen-hs** [reply]
sorry I am bothering you again :)
now both of the `pdm add` commands works, but after them, any `pdm add` of any library seems to fail because of the internal library
does it mean it forces me to always add dependencies to a specific lock file ?

```
pdm add -e $HSCODE/gimme --dev -L pdm.dev.lock                              
Adding group dev to lockfile
Adding packages to dev dev-dependencies: -e file:///Users/liran/hiredscore/git/gimme#egg=hs-gimme
🔒 Lock successful
```

and after that:
```
pdm add requests                                                                          
Adding packages to default dependencies: requests
🔒 Lock failed
Unable to find a resolution for hs-gimme
because of the following conflicts:
  -e file:///Users/liran/hiredscore/git/gimme#egg=hs-gimme (from project)
  hs-gimme @ git+ssh://git@github.com/HiredScore/gimme.git@63.4.65 (from project)
```

thanks again for the quick responses ! much appreciated

**Reply by frostming** [reply]
The issue is that now your `pdm.lock` **can't** contain the `dev` group(listed at `[metadata]` table`), then the local version won't ever show again.
